### Comparing Probability of survival using Bayes' Theorem  

a boy in 2nd class

a woman in 3rd class 

a man in 1st class

In [2]:
import pandas as pd

Location= "titanic3.xlsx"
df= pd.read_excel(Location)

In [3]:
df.describe()

pclass     survived          age        sibsp        parch  \
count  1309.000000  1309.000000  1046.000000  1309.000000  1309.000000   
mean      2.294882     0.381971    29.881135     0.498854     0.385027   
std       0.837836     0.486055    14.413500     1.041658     0.865560   
min       1.000000     0.000000     0.166700     0.000000     0.000000   
25%       2.000000     0.000000    21.000000     0.000000     0.000000   
50%       3.000000     0.000000    28.000000     0.000000     0.000000   
75%       3.000000     1.000000    39.000000     1.000000     0.000000   
max       3.000000     1.000000    80.000000     8.000000     9.000000   

              fare        body  
count  1308.000000  121.000000  
mean     33.295479  160.809917  
std      51.758668   97.696922  
min       0.000000    1.000000  
25%       7.895800   72.000000  
50%      14.454200  155.000000  
75%      31.275000  256.000000  
max     512.329200  328.000000

### a boy in 2nd class

In [4]:
def findbayessurvived(ismale,isbelow18,issecondclass,issurvived):
    return ismale*isbelow18*issecondclass*issurvived

In [5]:
len(df[df["sex"]== "male"]) & len(df[df['survived']== 1])

320

In [6]:
len(df[df["age"]< 18]) & len(df[df['survived']== 1])

144

In [7]:
len(df[df['pclass']== 2]) & len(df[df['survived']== 1])

276

In [8]:
len(df[df['survived']==1])

500

In [9]:
P_male_survived=320/500
P_below18_survived= 144/500
P_2ndclass_survived= 276/500
P_survived= 500/1309

print(findbayessurvived(P_male_survived,P_below18_survived,P_2ndclass_survived,P_survived))

0.03886349885408709


### a woman in 3rd class

In [10]:
len(df[df["sex"]=="female"]) & len(df[df['survived']==1])

464

In [11]:
len(df[df["age"]>= 18]) & len(df[df['survived']== 1])

372

In [12]:
len(df[df['pclass']== 3]) & len(df[df['survived']== 1])

196

In [18]:
P_female_survived=464/500
P_aboveor18_survived= 372/500
P_3rdclass_survived= 196/500
P_survived= 500/1309

print(findbayessurvived(P_female_survived,P_aboveor18_survived,P_3rdclass_survived,P_survived))

0.10338019251336898


### a man in 1st class

In [19]:
len(df[df["sex"]== "male"]) & len(df[df['survived']== 1])

320

In [15]:
len(df[df["age"]>= 18]) & len(df[df['survived']== 1])

372

In [16]:
len(df[df['pclass']== 1]) & len(df[df['survived']== 1])

320

In [17]:
P_male_survived=320/500
P_aboveor18_survived= 372/500
P_firstclass_survived= 320/500
P_survived= 500/1309

print(findbayessurvived(P_male_survived,P_aboveor18_survived,P_firstclass_survived,P_survived))

0.11640275019098549


Probability of survival for

a boy in 2nd class = 0.03886349885408709

a woman in 3rd class=0.10338019251336898

a man in 1st class= 0.11640275019098549
 
Therefore, among the three given groups the probability of survival is the highest for a man in 1st class, following by a woman in 3rd class; and a boy in 2nd class have the lowest probability of survival.